src/Engines/thinEngine.ts

```diff
@@ -675,6 +676,7 @@ export class ThinEngine {
                         // Cache
                         this.wipeCaches(true);
                         Logger.Warn("WebGL context successfully restored.");
+                        this.onContextRestoreBuffersObservable.notifyObservers(this);

```

Comment:
Why not using onContextRestoredObservable ?

Reply:
This is just a thought I had. If we move when they are created from a well-defined point before onContextRestoredObservable to onContextRestoredObservable then rebuilds of buffers will happen randomly intercepted with whatever the user might want to do at a rebuild. I was thinking that if the user did something that required the inner buffers to actually exist, or if she did something that forced a premature/extra rebuild of the buffers. That could cause confusion and maybe bugs.

I'm thinking that thinEngine could have an array of all buffers and just loop and update them (if it supports lost context). That is probably more efficient compared to having an observer at all.. But I could be wrong on that point.

If is very easy to move everything over to onContextRestoredObservable again of course if we feel that it is sufficient.

Reply:
I don't think this is an issue. Let's not change too many things at once ;)
I would like to merge it asap because we have too many moving pieces